src/azul/plugins/metadata/hca/transform.py

```diff
@@ -341,26 +344,27 @@ def __init__(self, id: str, title: str, category: SubmitterCategory):
         self.slug = slug
         self.title = title
         self.category = category
-        self.by_id[self.id] = self
+        assert title not in self.by_title, title
+        self.by_title[title] = self
+        self.by_id[id] = self
 
     @classmethod
-    def for_id(cls, submitter_id: str) -> Optional['Submitter']:
+    def for_id(cls, id: str) -> Optional['Submitter']:
```

Comment:
In the constructor, it's pretty clear that the `id` argument refers to a submitter ID. Especially because the attribute is also called `id`. In this factory, not so much. Please revert.